### ISSUE TITLE
Containerize this Project for Easier Contribution

### ISSUE THREAD

**Issue by sujal-goswami** [wavetermdev/waveterm#740]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when trying to set up the development environment for this project. The process can be time-consuming and error-prone, particularly when dealing with different dependencies and system configurations.

**Describe the solution you'd like**
I propose containerizing this project using Docker. By providing a pre-configured Docker container, developers can quickly and easily set up a consistent development environment. This would streamline the onboarding process for new contributors and ensure that everyone is working with the same setup, reducing the likelihood of environment-related issues.

**Comment by esimkowitz** [comment]
Good idea, our dependencies are pretty straightforward if you want to give it a go!

**Comment by sujal-goswami** [comment]
I have given it a try, but unfortunately, it resulted in some unusual errors. Could you please take a look at it?

![image](https://github.com/user-attachments/assets/e12e9762-0e0b-471d-a0d3-87c2226f9423)
![image](https://github.com/user-attachments/assets/59b48aec-ce78-401d-bbfe-391d3385b86e)
![image](https://github.com/user-attachments/assets/e4c9051c-f6a0-471e-80c3-8f6af8922541)
![image](https://github.com/user-attachments/assets/7e0fd1aa-c875-46d9-9ccd-752206a77ae3)

**Comment by esimkowitz** [comment]
Wavesrv depends on waveshell, so you can't copy just wavesrv when creating the wavesrv-builder stage, I would recommend running as a monolithic backend-builder

**Comment by sujal-goswami** [comment]
Hey, during my local build, I discovered some issue in build-linux.md in there `npm install -g yarn` is installing other version of yarn but this project uses version` "yarn@4.1.1"` which causing problem.
![image](https://github.com/user-attachments/assets/a0fc869a-078f-4982-8286-cc157fcb2f53)

but even after resolving it during running client, it gives a strange error 
![image](https://github.com/user-attachments/assets/dabdfc83-408c-4110-8419-5fa5f7db290f)

**Comment by esimkowitz** [comment]
Ah, it looks like our documentation in the build-linux document is out of date, we use Yarn Modern for our package manager, which requires some extra setup. Check out the info in the macOS instructions regarding Yarn: https://github.com/wavetermdev/waveterm/blob/main/BUILD.md#prereqs-and-tools

Specifically, you need to run the following:

```sh
corepack enable
yarn
```

**Comment by osallak** [comment]
Hello @esimkowitz ,  can i work on this?

**Comment by esimkowitz** [comment]
Ah that was a batch job i ran when we released 0.8 to label all older issues as legacy, since that was a full rewrite and most of the old issues were no longer relevant. As for whether this is still relevant, that's kinda tough. There was an open PR from @sujal-goswami to try and address this but it was still a lot of setup because you need to pipe through a graphical environment to actually launch the app and display it in your local env so you had to have X server installed and setup locally. We've also significantly reduced our build dependencies since this issue was created. Packaging the app still requires a good number of dependencies, but building should only require Go, Node, and Task on macOS and Go, Node, Task, and Zig on Windows and Linux.

**Comment by esimkowitz** [comment]
in my view, dev containers work well for services and anything that is usually represented as a containerized application, but it's not a great way to test a user application, since it restricts your ability to interact with the actual system. We could look into setting up a Nix flake or something to simplify dependency management for our project, but I think it's still best to test on an actual machine.

**Comment by osallak** [comment]
Thanks for the detailed clarification, @esimkowitz  I'm highly interested in contributing to this project. If you could point me towards some good first issues that are currently relevant, that would be greatly appreciated!
